keps/0020-manual-plan-execution.md

```diff
@@ -0,0 +1,175 @@
+
+## Summary
+
+Since the inception of KUDO, being able to manually run a plan was one of the core features. After KEP-18 (controller overhaul) we changed the way we store information about executed plans practically removing the ability to execute custom plan in an easy way through kubectl. Goal of this KEP is to propose a way how to re-introduce that functionality and provide a good user experience.
+
+## Motivation
+
+### Goals
+
+Operator will be able to trigger plan of given name through CLI
+Controller won’t allow multiple plans running in parallel
+
+### Non-Goals
+
+Provide a history of plan executions older that the one last run per plan.
+
+## Proposal
+
+There are multiple ways we can do this. The goal of the KEP is to choose the one with the right balance of added complexity and with a good enough UX. The main challenge with executing plans is that kubernetes API in its essence is declarative while starting a plan is an imperative action (a command) so it does not fit very nicely into the ecosystem.
+
+### Update the status of Instance with name of the plan you want to run, let controller pick it up
+
+This is by far the simplest solution but probably not the one with cleanest design. The implementation of this could be as simple as introducing new CLI command that would do:
+
+```go
+newStatus := instance.Status.DeepCopy()
+// TODO planStatus now contains status of that plan we want to run
+// for implementation details of that look at instance.StartPlanExecution
+planStatus.Status = ExecutionPending
+planStatus.UID = uuid.NewUUID()
+
+instance.Status = newStatus
+client.Status().Update(context.TODO(), instance)
+```
+
+This also counts with the fact that on the server-side we have an admission webhook that won't allow setting a status like that in case there is another plan running. Such update would be rejected.
+
+Although it’s very easy to do this, it’s not very kubernetes idiomatic way of doing things especially because Status should never be updated from a client and it should just capture the state of the object.
+
+For some background, including definition of status subresource by [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status): `The status summarizes the current state of the object in the system, and is usually persisted with the object by an automated processes but may be generated on the fly. At some cost and perhaps some temporary degradation in behavior, the status could be reconstructed by observation if it were lost.`  
+
+#### Pros
+
+Very easy to implement considering the current state of the code
+
+#### Cons
+
+No way to start a plan with just pure kubectl - this is a reply from sig-cli channel where I posed that question (`It doesn't make sense to update the status manually, it should reflect the current state in the cluster. So no, you cannot set it though kubectl`)
+
```

Comment:
if we are counting preferences... I'm against this first approach for reasons mentioned in the Cons list.